Assets/Scripts/Level/Level.cs

```diff
@@ -206,6 +217,19 @@ void Update()
             if (spawnQueue.Count > 0 && nextSpawn == null)
             {
                 nextSpawn = spawnQueue.Dequeue();
+
+                Debug.Log(enemySpawnedCounter);
+
+                // This has to be above the if check
+                enemySpawnedCounter += 1;
+
+                // Play next song if applicable
+                if (enemySpawnedCounter == playSongWhenSpawnEnemyCount[0])
```

Comment:
I get what this is doing, but I'm a bit unclear on why we'd do this rather than just transition based on track length. Would you mind giving a brief overview of the reasoning behind this approach?

Reply:
Tracks that loop requires manually calling `PlayNextSection` to escape from loop.  Tracks that do not loop will automatically play. So this logical here is to make sure that we exit out of the loop. Hope that helps.

Reply:
Gotcha!